ts/Dashboards/Component/KPIComponent.ts

```diff
@@ -349,11 +349,73 @@ class KPIComponent extends Component {
         }
         return '';
     }
+    public toJSON(): KPIComponent.ClassJSON {
+        const base = super.toJSON();
+        const json = {
+            ...base,
+            options: {
+                ...base.options,
+                value: this.options.value,
+                subtitle: JSON.stringify(this.options.subtitle),
+                title: JSON.stringify(this.options.title),
+                threshold: this.options.threshold,
+                style: JSON.stringify(this.options.style),
+                thresholdColors: this.options.thresholdColors,
+                chartOptions: JSON.stringify(this.options.chartOptions),
+                valueFormat: this.options.valueFormat
```

Comment:
Why we cannot `stringify` the whole `this.options` object? The value can be string too.

Reply:
valueFormatter and callback functions in general should not be stringified for security reasons as it opens the door to inject code via JSON.

Reply:
I can't stringify `valueFormatter` because it is a callback function.  `valueFormat` on the other hand is a string, so it can work, as you said :D

Reply:
Additionally, I needed to merge the options with the base json.